main.py

```diff
@@ -1,16 +1,236 @@
+
+        Returns:
+            returns dictionary d that represents the object attributes
+                """
+        print(self.id_, self.first_name, self.last_name, self.age, self.gender, self.address, self.phone_numbers)
+
+        dict_user = {"id": self.id_, "first_name": self.first_name, "last_name": self.last_name, "age": self.age,
+                     "gender": self.gender, "address": self.address, "phone_numbers": self.phone_numbers}
+        # print(dict_user)
+        return dict_user
+
+
+def add_to_class(temp: object) -> bool:
+    """this function aims to preserve the uniqueness of the id
+            Args:
+                temp: which represent the temp object of the user
+
+            Returns:
+                true if id is unique, false if not
+
+            """
+    for x in users_object_list:
+        if temp.id_ == x.id_:
+            return False
+    users_object_list.append(temp)
+    return True
+
+
+def get_users_data() -> dict:
+    """ the function aime to read the data from the input file
+    Args:
+        this function doesn't need any arguments
+    Returns:
+        The return value is a dictionary of the input data
+    """
+
+    with open('users.json') as f:
+        users_temp = json.load(f)
+    return users_temp
+
+
+def create_users_list(users_data: dict) -> list:
+    """ the function aime to create a list of objects for each user from the input file
+    Args:
+        users_data (dict): represent the input data
+    Returns:
+        The return value is a list of the input objects
+    """
+    user_objects = []
+    for i in users_data:
+        temp = User(**i)
+        # temp = User(i["id"], i["first_name"], i["last_name"], i["gender"], i["age"], i["address"], i["phone_numbers"])
+        user_objects.append(temp)
+    return user_objects
+
+
+@app.route('/users', methods=['GET', 'POST'])
+def users() -> json:
+    """this function performs both get and post methods
+
+        Args:
+            there is no parameters
+
+        Returns:
+            returns a json that contains all the wanted result plus a 201 status code
+
+        """
+    if request.method == 'GET':
+        users_names = get_users()
+        return jsonify(users_names), 201
+
+    elif request.method == 'POST':
+        result = add_user()
+        return result, 201
+
+
+@app.route('/users/<int:user_id>', methods=['GET', 'PUT', 'DELETE'])
+def users_with_id(user_id: int) -> json:
+    """this function performs the get, put and delete operation on the user with the given id
+
+        Args:
+            user_id (int): represents the id_ of the user we want to perform the operation on
+
+        Returns:
+            this function returns a json the either states if the operation is succesfull or not or the data of the user
+
+        """
+    if request.method == 'GET':
+        result = get_user_with_id(int(user_id))
+        return jsonify(result), 201
+    elif request.method == 'DELETE':
```

Comment:
what if the user is not found?
you must return different code then, also the code should be here 200